homeassistant/components/peblar/update.py

```diff
@@ -0,0 +1,89 @@
+"""Support for Peblar updates."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.update import UpdateEntity, UpdateEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import (
+    PeblarConfigEntry,
+    PeblarVersionDataUpdateCoordinator,
+    PeblarVersionInformation,
+)
+
+
+@dataclass(frozen=True, kw_only=True)
+class PeblarUpdateEntityDescription(UpdateEntityDescription):
+    """Describe an Peblar update entity."""
+
+    installed_fn: Callable[[PeblarVersionInformation], str | None]
+    available_fn: Callable[[PeblarVersionInformation], str | None]
+
+
+DESCRIPTIONS: tuple[PeblarUpdateEntityDescription, ...] = (
+    PeblarUpdateEntityDescription(
+        key="firmware",
+        translation_key="firmware",
```

Comment:
Can get the Firmware device class and can remove the translation key

Reply:
Update entities are not eligible for that.
That logic isn't applied to that platform.

Reply:
It is

Reply:
https://github.com/home-assistant/core/blob/6cf10cd0b221dc92a41dbdefd9e2580a1b856873/homeassistant/components/update/__init__.py#L250-L255

Reply:
Then it is broken... 👯

Reply:
Works on my machine?
https://github.com/home-assistant/core/blob/bd55fe868d232c2106503a78e6175fccff14d0e1/tests/components/airgradient/snapshots/test_update.ambr#L39-L42

Reply:
facepalm... device class is missing...